aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1689,5 +1690,89 @@ Tensor chain_matmul(TensorList matrices) {
   }
 }
 
+DEFINE_DISPATCH(unpack_pivots_stub);
+
+std::tuple<Tensor, Tensor, Tensor> _lu_unpack(
+    const Tensor& LU_data,
+    const Tensor& LU_pivots,
+    bool pivots_as_permutation_matrix
+    ) {
+  // In the generalized LU factorization, the following shape relations hold:
+  // A.shape[-2:] == (m, n),
+  // P.shape[-2:] == (m, m),
+  // U.shape[-2:] == (m, k),
+  // L.shape[-2:] == (k, n),
+  // where k = min(m, n)
+  int64_t m = LU_data.size(-2);
+  int64_t n = LU_data.size(-1);
+  int64_t k = std::min(m, n);
+
+  auto U = LU_data.triu();
+  if (m != k) {
+    U = U.narrow(-2, 0, k);
+  }
+
+  auto L = LU_data.tril();
+  if (k != n) {
+    L = L.narrow(-1, 0, k);
+  }
+  L.diagonal(/*offset=*/0, /*dim1=*/-2, /*dim2=*/-1).fill_(1);
+
+  auto unpacked_pivots_sizes = LU_pivots.sizes().vec();
+  unpacked_pivots_sizes[LU_pivots.dim() - 1] = m;
+  auto unpacked_pivots = at::zeros(
```

Comment:
This could be an `at::empty` as you write over all the values in the `copy_` below.

Reply:
`copy` does resize?